src/storage/src/source/postgres/replication.rs

```diff
@@ -355,21 +360,7 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                                         data_output.give(data_cap, update).await;
                                     }
                                 }
-                                assert!(
-                                    new_upper <= commit_lsn,
-                                    "new_upper={} tx_lsn={}",
-                                    new_upper,
-                                    commit_lsn
-                                );
-                                container.push((data, commit_lsn, diff));
-                            }
-                            new_upper = commit_lsn + 1;
-                            if container.len() > max_capacity {
-                                data_output
-                                    .give_container(&data_cap_set[0], &mut container)
-                                    .await;
-                                upper_cap_set.downgrade([&new_upper]);
-                                data_cap_set.downgrade([&new_upper]);
```

Comment:
shouldn't we still downgrade the `data_cap_set` capability after producing the last update for this transaction?

Reply:
There is no need to downgrade the capability until we are about to yield, which happens a few lines below and there the capability is downgraded once